pgjdbc/src/test/java/org/postgresql/test/jdbc2/TimestamptzTest.java

```diff
@@ -0,0 +1,239 @@
+  }
+
+  /**
+   * Aother example where overloading is necessary on sql site to choose the correct function
+   */
+  @Test
+  public void testTypeOnDbSite_functionOverloading() throws SQLException {
+
+    //SET time zone 'Europe/Berlin';
+    try (PreparedStatement ps = con.prepareStatement("create or replace function testtimestamp.demoF(msg text, ts timestamptz) returns text language sql as $$select msg || ' at time ' || ts $$;  ")) {
+      ps.executeUpdate();
+    }
+    try (PreparedStatement ps = con.prepareStatement("create or replace function testtimestamp.demoF(msg text, additional text) returns text language sql as $$select msg || ' wrongInfo:' || additional  $$; ")) {
+      ps.executeUpdate();
+    }
+
+    GregorianCalendar cal = new GregorianCalendar(TimeZone.getTimeZone("Europe/Berlin"));
+    cal.set(2023, Calendar.MARCH, 12, 9, 30);
+    cal.set(Calendar.SECOND, 0);
+    cal.set(Calendar.MILLISECOND, 0);
+
+    try (PreparedStatement ps = con.prepareStatement(" SELECT  testtimestamp.demoF(?, ?) as res ")) {
+      ps.setString(1, "Some data");
+      ps.setTimestamp(2, new Timestamp(cal.getTimeInMillis()));
+
+      ResultSet rs = ps.executeQuery();
+      if (rs.next()) {
+        String resultFunction = rs.getString("res");
+        System.out.printf("result:%s \n", resultFunction);
+
+        if (resultFunction.contains("wrongInfo")) {
+          fail("choose wrong overloading");
+        }
+
+      } else {
+        fail("no result");
+      }
+    }
+  }
+
+
+
+  /**
+   * Test to demonstrate another example on additional unnecessary cast.
+   * The driver know already the data type.
+   */
+  @Test
+  public void testTypeOnDbSite_select() throws SQLException {
+
+    //SET time zone 'Europe/Berlin';
+    try (PreparedStatement ps = con.prepareStatement(" insert into testtimestamp.tbtesttimestamp (id, ts) values(1, '2023-03-12 10:00:00+1'::timestamptz ) ")) {
+      ps.executeUpdate();
+    }
+
+    GregorianCalendar cal = new GregorianCalendar(TimeZone.getTimeZone("Europe/Berlin"));
+    cal.set(2023, Calendar.MARCH, 12, 9, 30);
+    cal.set(Calendar.SECOND, 0);
+    cal.set(Calendar.MILLISECOND, 0);
+
+    try (PreparedStatement ps = con.prepareStatement(" SELECT * from testtimestamp.tbtesttimestamp where ts < ?  +  interval '1 hour' ")) {
+      ps.setTimestamp(1, new Timestamp(cal.getTimeInMillis()));
+
+      ResultSet rs = ps.executeQuery();
+      if (rs.next()) {
+        System.out.printf("id:%d ts:%s \n", rs.getInt("id"), rs.getTimestamp("ts"));
+      } else {
+        fail("no result");
+      }
+    }
+  }
+
+
+  /**
+   * Add test for null and correct value of null parameter type.
+   * The test should work with and without TimestamptzAlways
+   */
+  @Test
+  public void testTypeOnDbSite_select_nullability() throws SQLException {
+
+    try (PreparedStatement ps = con.prepareStatement("select 1 where ? is null  ")) {
+      ps.setTimestamp(1, null);
+      ResultSet rs = ps.executeQuery();
```

Comment:
Should check the result. What if the result set is empty ?

Reply:
the check is not about the result but more on the syntax, without the fix postgres return:
org.postgresql.util.PSQLException: ERROR: could not determine data type of parameter $1

Reply:
added check for non empty result set